src/collection.js

```diff
@@ -464,9 +464,9 @@ export default class Collection {
       return reject("Extraneous Id; can't create a record having one set.");
     }
     const newRecord = {...record,
-      id:      options.synced ||
-                   options.useRecordId ? record.id :
-                                     this.idSchema.generate(),
+      id: options.synced ||
+                       options.useRecordId ? record.id :
+                       this.idSchema.generate(record),
```

Comment:
When you mention using an intermediary variable, are you referring to the `id` or the `record` being passed to the generate command?